ui/apps/platform/src/Components/SelectSingle/SelectSingle.tsx

```diff
@@ -38,33 +41,75 @@ function SelectSingle({
 }: SelectSingleProps): ReactElement {
     const [isOpen, setIsOpen] = useState(false);
 
-    function onSelect(_event, selection) {
-        // The mouse event is not useful.
-        setIsOpen(false);
-        handleSelect(id, selection);
+    function onSelect(
+        _event: React.MouseEvent<Element, MouseEvent> | undefined,
+        selection: string | number | undefined
+    ) {
+        if (typeof selection === 'string') {
+            setIsOpen(false);
+            handleSelect(id, selection);
+        }
     }
 
+    function onToggle() {
+        setIsOpen(!isOpen);
+    }
+
+    // Find the display text for the selected value
+    const getDisplayText = (): string => {
+        if (!value) {
+            return placeholderText;
+        }
+
+        const selectedChild = children.find((child) => {
+            return child.props.value === value;
+        });
+
+        return (selectedChild?.props.children as string) || value;
+    };
+
+    const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+        <MenuToggle
+            ref={toggleRef}
+            onClick={onToggle}
+            isExpanded={isOpen}
+            isDisabled={isDisabled}
+            icon={toggleIcon}
+            aria-label={toggleAriaLabel}
+            id={id}
+            variant="default"
+            className="pf-v5-u-w-100"
+        >
+            {getDisplayText()}
+        </MenuToggle>
+    );
+
     return (
         <Select
-            variant={variant === 'typeahead' ? 'typeahead' : 'single'}
-            toggleIcon={toggleIcon}
-            toggleAriaLabel={toggleAriaLabel}
             id={id}
-            isDisabled={isDisabled}
+            aria-label={toggleAriaLabel}
             isOpen={isOpen}
+            selected={value}
             onSelect={onSelect}
-            onToggle={(_event, val) => setIsOpen(val)}
-            selections={value}
-            direction={direction}
-            isCreatable={isCreatable}
-            placeholderText={placeholderText}
-            toggleId={id}
+            onOpenChange={(nextOpen: boolean) => setIsOpen(nextOpen)}
+            toggle={toggle}
+            shouldFocusToggleOnSelect
+            popperProps={
+                menuAppendTo
+                    ? {
+                          appendTo: menuAppendTo,
```

Comment:
Small nit: if `menuAppendTo` is undefined, does the behavior change versus omitting the property entirely? If not, we could save the ternary and use a single object here.